pallets/parachain-staking/src/migrations.rs

```diff
@@ -47,6 +47,43 @@ where
 	}
 }
 
+/// Compute RoundInfo theoretical first_slot
+pub struct ComputeTheoreticalFirstSlot<T: Config>(core::marker::PhantomData<T>);
+
+impl<T> OnRuntimeUpgrade for ComputeTheoreticalFirstSlot<T>
+where
+	T: Config,
+	BlockNumberFor<T>: From<u32> + Into<u64>,
+{
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+		Ok(Vec::new())
+	}
+
+	fn on_runtime_upgrade() -> frame_support::pallet_prelude::Weight {
+		// Read round
+		let mut round = crate::Round::<T>::get();
+
+		// Compute theoretical `first_slot``
+		round.first_slot = compute_theoretical_first_slot(
+			<frame_system::Pallet<T>>::block_number(),
+			round.first,
+			u64::from(T::SlotProvider::get()),
+			T::BlockTime::get(),
+		);
+
```

Comment:
Just to confirm. We do not need to check `DelayedPayouts` in this migration?

Reply:
`DelayedPayouts` was already migrated with the migration `MigrateRoundWithFirstSlot` that was applied to all runtimes at rt 28xx (this migration was not enoug for moonriver/moonbeam because the theoretical first slot was computed with a SlotProvider that use the parachain block instead of the relay slot